### PR TITLE
Use thermo state field cache more

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds more use of the thermo state field cache. I think this is refactoring for the most part, but there is one behavioral change in `src/EDMF_Environment.jl`. @trontrytel, can you take a look and confirm that this looks okay? (the cached thermo state was constructed using the if-else block, and the one replaced does not)

### CODE REVIEW

File: post_processing/mse_tables.jl [hunk 1]
```diff
@@ -112,20 +112,20 @@ all_best_mse["life_cycle_Tan2018"]["updraft_thetal"] = 0.0
 all_best_mse["life_cycle_Tan2018"]["v_mean"] = 0.0
 all_best_mse["life_cycle_Tan2018"]["u_mean"] = 0.0
 all_best_mse["life_cycle_Tan2018"]["tke_mean"] = 0.0
-all_best_mse["life_cycle_Tan2018"]["temperature_mean"] = 6.285660087520289e-18
+all_best_mse["life_cycle_Tan2018"]["temperature_mean"] = 0.0
```
Comment (trontrytel) [inline post_processing/mse_tables.jl:115]: Why do we now have so many zeros in the mses?

File: src/update_aux.jl [hunk 2]
```diff
@@ -146,30 +148,15 @@ function update_aux!(edmf::EDMFModel, grid::Grid, state::State, surf::SurfaceBas
...
-                aux_en.q_tot[k],
-                aux_en.q_liq[k],
-                aux_en.q_ice[k],
-            )
+            (aux_en.q_liq[k], aux_en.q_ice[k])
         else
             error("Something went wrong. The moisture_model options are equilibrium or nonequilibrium")
         end
-        aux_gm.θ_virt[k] = TD.virtual_pottemp(param_set, ts_gm)
+        ts_env[k] = thermo_state_pθq(param_set, p0_c[k], aux_en.θ_liq_ice[k], aux_en.q_tot[k], thermo_args...)
```
Comment (trontrytel) [inline src/update_aux.jl:158]: very cool

**Comment by trontrytel** [comment]
>the cached thermo state was constructed using the if-else block, and the one replaced does not

Not sure I see the thing you are asking about @charleskawczynski ? Seems like there was an if/else due to moisture model in the environment?

File: src/EDMF_Environment.jl [hunk 3]
```diff
@@ -323,7 +317,7 @@ function microphysics(
 
         else
             # if variance and covariance are zero do the same as in SA_mean
-            ts = thermo_state_pθq(param_set, p0_c[k], aux_en.θ_liq_ice[k], aux_en.q_tot[k])
+            ts = ts_env[k]
```
Comment (charleskawczynski) [inline src/EDMF_Environment.jl:320]: @trontrytel this is the line that is technically behavior changing. This doesn’t have equil-nonequil branches

Comment (trontrytel) [reply src/EDMF_Environment.jl:320]: Right. It doesn't matter. We don't support the non-equilibrium option with the sgs quadratures, so the user will get an error somewhere before that line. 

I'll add the docs soon.

**Review by trontrytel — APPROVED** [review]
LGTM. The only sus part are the zeros in the mse tables. And why there are so many changes there in the first place

File: post_processing/mse_tables.jl (hunk 1 above)
Comment (charleskawczynski) [reply post_processing/mse_tables.jl:115]: This was a head scratcher. Then I remembered, we have a precedence order for comparing datasets, and TC main comes [second](https://github.com/CliMA/TurbulenceConvection.jl/blob/main/post_processing/compute_mse.jl#L445) after LES (and we don’t have LES for life cycle). We should simplify our regression tests, or make them clearer somehow.. one thing at a time.

**Comment by charleskawczynski** [comment]
I can explain the many changes tomorrow, it’s a bit confusing..

**Comment by charleskawczynski** [comment]
But unrelated to this PR

**Comment by charleskawczynski** [comment]
Bors r+

**Pull request merged**